src/lib/sql/sql_pipeline.cpp

```diff
@@ -145,16 +145,19 @@ const std::vector<std::shared_ptr<AbstractLQPNode>>& SQLPipeline::get_optimized_
          "One or more SQL statement is dependent on the execution of a previous one. "
          "Cannot translate all statements without executing, i.e. calling get_result_table()");
 
+  // The optimizer works on the original unoptimized LQP nodes. After optimizing, the unoptimized version is also
```

Comment:
I think I understand what you want to say but I find it a bit confusing. Maybe it gets clearer if we just say that the unoptimized version is altered during the optimization process. 
A different question, why do we have to modify the unoptimized version at all, can't we just create a copy if we have `_optimized_logical_plans` anyways?

Reply:
```
// The optimizer modifies the input LQP and requires exclusive ownership of that LQP. This means that we need to clear
// _unoptimized_logical_plans. This is not an issue as the unoptimized plans will no longer be needed. Note that this
// does not clear the unoptimized LQPs stored in the SQLPipelineStatement - those are cleared as part of
// SQLPipelineStatement::get_optimized_logical_plan. Calls to get_unoptimized_logical_plans are still allowed (e.g.,
// for visualization), in which case the unoptimized plan will be recreated.
```

better?

Reply:
If yes, I'll adapt the version in sql_pipeline_statement.cpp accordingly

Reply:
Yes